sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -1579,6 +1580,49 @@ def test_lull_logging(self):
         '.*There has been a processing lull of over.*',
         'Unable to find a lull logged for this job.')
 
+class StateBackedTestElementType(object):
+  element_count = 0
+
+  def __init__(self, num_elems):
+    self.num_elems = num_elems
+    self.value = ['a' for _ in range(num_elems)]
+    StateBackedTestElementType.element_count += 1
+    # Due to using state backed iterable, we expect there is a few instances
+    # alive at any given time.
+    if StateBackedTestElementType.element_count > 5:
+      raise RuntimeError('Too many live instances.')
+
+  def __del__(self):
+    StateBackedTestElementType.element_count -= 1
+
+  def __reduce__(self):
+    return (self.__class__, (self.num_elems, ))
+
+@attr('ValidatesRunner')
+class FnApiBasedStateBackedCoderTest(unittest.TestCase):
+
+  class ElementDoFn(beam.DoFn):
+    def process(self, elements):
+      unused_key, ts = elements
+
+      yield sum([item.num_elems for item in ts])
+
+  def create_pipeline(self):
+    return beam.Pipeline(
```

Comment:
We'd need TestPipeline to be a ValidatesRunner test. Of course in that case we couldn't manually pass use_state_iterables (unless we make it a pipeline option).

However, I'm +1 for this test going in and then future work getting it to run on other runners (possibly just overriding the create_pipeline method altogether).

Reply:
acked.